### ISSUE TITLE
Reporting some minor bugs in visual studio

### ISSUE THREAD

**Issue by rezathecoder** [dotnet/roslyn#49965]
### 1.Intellisense in watch window
IntelliSense automatically selects suggestion when enter is hit. In earlier versions of vs you would hit enter and it completed the code and you could write rest of the code but now it looks like you hit enter twice 

![1](https://user-images.githubusercontent.com/66854399/101995840-3b4b5400-3ce2-11eb-9012-9b8ed1c4b1d1.gif)

### 2.Rename problem in views
Visual Studio is not renaming types, properties, ... referenced in view files if they're not open

![2](https://user-images.githubusercontent.com/66854399/101995903-f1af3900-3ce2-11eb-8ab1-803b6480d2ce.gif)
 
### 4. Weird output when renaming types

Sometimes rename operation produces weird code in views and it will ruin your view. Output is something like:

```
#line default
#line hidden
#nullable disable
                await __tagHelperRunner.RunAsync(__tagHelperExecutionContext);
#nullable restore
```

### 5.IntelliSense suggestion problem
IntelliSense will not suggest un-imported types (types that do not have using in current document) after a while passed from installation time.

![3](https://user-images.githubusercontent.com/66854399/101995985-b06b5900-3ce3-11eb-8702-897ab377003f.gif)

### 6.ArgumentOutOfRangeException when writing c# code in views
I get this exception a lot when i start to write c# code inside razor views specially when i just started working in visual studio. Looks like it happens when it tries to format my view. Log:

```
System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. Parameter name: lengthat Microsoft.VisualStudio.Text.Span..ctor(Int32 start, Int32 length)at Microsoft.VisualStudio.LanguageServices.Implementation.Venus.ContainedDocument.AdjustSpan(Span span, List`1 replacementMap)at Microsoft.VisualStudio.LanguageServices.Implementation.Venus.ContainedDocument.<GetSubTextChanges>d__56.MoveNext()at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)at Microsoft.VisualStudio.LanguageServices.Implementation.Venus.ContainedDocument.<FilterTextChanges>d__52.MoveNext()at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)at Microsoft.VisualStudio.LanguageServices.Implementation.Venus.ContainedDocument.UpdateText(SourceText newText)at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioWorkspaceImpl.ApplyTextDocumentChange(DocumentId documentId, SourceText newText)at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioWorkspaceImpl.ApplyDocumentTextChanged(DocumentId documentId, SourceText newText)at Microsoft.CodeAnalysis.Workspace.ApplyChangedDocument(ProjectChanges projectChanges, DocumentId documentId)at Microsoft.CodeAnalysis.Workspace.ApplyProjectChanges(ProjectChanges projectChanges)at Microsoft.CodeAnalysis.Workspace.TryApplyChanges(Solution newSolution, IProgressTracker progressTracker)at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioWorkspaceImpl.TryApplyChanges(Solution newSolution, IProgressTracker progressTracker)at Microsoft.CodeAnalysis.Workspace.TryApplyChanges(Solution newSolution)at Microsoft.CodeAnalysis.Editor.Implementation.AutomaticCompletion.Extensions.ApplyTextChanges(Document document, IEnumerable`1 textChanges, CancellationToken cancellationToken)at Microsoft.CodeAnalysis.Editor.CSharp.AutomaticCompletion.Sessions.CurlyBraceCompletionSession.AfterReturn(IBraceCompletionSession session, CancellationToken cancellationToken)at Microsoft.CodeAnalysis.Editor.Implementation.AutomaticCompletion.BraceCompletionSessionProvider.BraceCompletionSession.PostReturn()at Microsoft.VisualStudio.Text.BraceCompletion.Implementation.BraceCompletionManager.<PostReturn>b__29_0()at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call) --- End of stack trace from previous location where exception was thrown --- at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

### Further technical details
- ASP.NET Core version: 3.1.10
- Visual Studio: 16.8.3

**Comment by NTaylorMullen** [comment]
@rezathecoder Thank you so much for all of your detailed gifs! For your 3rd item I deleted the content since I transferred the issue to Roslyn for the rest. In regards to it though try out our new experimental Razor editor, it should be resolved there (Tools -> Options -> Environment -> Preview Features):

![image](https://user-images.githubusercontent.com/2008729/87990285-8b5a1d80-ca98-11ea-914f-949fa326d9e3.png)

I'll let the Roslyn folks comment on the rest. I imagine they may want to break this issue up into multiple parts / will need additional info but I'll leave it to them to follow up.

**Comment by rezathecoder** [comment]
@NTaylorMullen Thank you for answering. I enabled experimenal razor editor and as shown in following gif i'm facing new issues:
1.`@` symbol is no working. I can't switch to c#
2.In old editor when you click on the name of variable `test` all the other occurrences would be highlighted but in experimental editor it's gone.
3.Rename operation does not even rename variables outside `<text>` tag.

![MsITUQmDkb](https://user-images.githubusercontent.com/66854399/102486296-15c1a000-407e-11eb-960e-375bcd52dbf2.gif)

Thank you for looking into this and i'm waiting for Roslyn folks to comment on other issues.

**Comment by rezathecoder** [comment]
@NTaylorMullen I also discovered another issue. So i disabled the experimental razor editor since it did not resolve my problem, I noticed that for the first time you run rename operation after visual studio startup following exception will be thrown:

`System.InvalidOperationException: Sequence contains more than one elementat System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.OpenTextBufferManager.OnTextBufferChanged(Object sender, TextContentChangedEventArgs args)at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.RaiseEvent[TArgs](Object sender, EventHandler`1 eventHandlers, TArgs args) --- End of stack trace from previous location where exception was thrown ---at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)`

![AaaSWvs4KK](https://user-images.githubusercontent.com/66854399/102487384-bb294380-407f-11eb-8570-4c78452fc4b0.gif)

This happens only once and if you want to reproduce you have to close visual studio completely and start again.

**Comment by NTaylorMullen** [comment]
> 1.`@` symbol is not working. I can't switch to c# when inside `<script>` tag
> 2.In old editor when you click on the name of variable `test` all the other occurrences would be highlighted but in experimental editor it's gone.
> 3.Rename operation does not even rename variables outside `<text>` tag.

Thanks for following up! These should have all been fixed in 16.9 preview builds, wooo! I believe 16.9-Preview2+ included the rename fix for JS (@alexgav ?). Prior to 16.9 the experimental Razor editor didn't have JS support.

> @NTaylorMullen I also discovered another issue. So i disabled the experimental razor editor since it did not resolve my problem, I noticed that for the first time you run rename operation after visual studio startup following exception will be thrown:

I'd bet that's fixed in 16.9-Preview2 in the new editor 😄

**Comment by CyrusNajmabadi** [comment]
Closing out.  Please file issues separately if you are still running into them.  Many of these are unrelated to roslyn,a nd having a single issue for all of these prevents being able to suitably move things around.
